lib/Doctrine/Persistence/Mapping/RuntimeReflectionService.php

```diff
@@ -3,18 +3,30 @@
 namespace Doctrine\Persistence\Mapping;
 
 use Doctrine\Common\Reflection\RuntimePublicReflectionProperty;
+use Doctrine\Common\Reflection\TypedNoDefaultReflectionProperty;
 use ReflectionClass;
 use ReflectionException;
 use ReflectionMethod;
 use ReflectionProperty;
+use function array_key_exists;
 use function class_exists;
 use function class_parents;
+use function phpversion;
+use function version_compare;
 
 /**
  * PHP Runtime Reflection Service.
  */
 class RuntimeReflectionService implements ReflectionService
 {
+    /** @var bool */
+    private $supportsTypedPropertiesWorkaround;
+
+    public function __construct()
+    {
+        $this->supportsTypedPropertiesWorkaround = version_compare((string) phpversion(), '7.4.0') >= 0;
```

Comment:
I suggest replacing this property with an inline `\PHP_VERSION_ID > 70400` check, which will be resolved at compile time by OPCache.